packages/botonic-api/src/rest/routes/events.ts

```diff
@@ -1,38 +1,177 @@
+import { BotonicEvent } from '@botonic/core/src/models/events'
 import { Router } from 'express'
+import { checkSchema, matchedData, validationResult } from 'express-validator'
+
+import { dataProviderFactory } from '../../data-provider'
+import {
+  eventIdParamSchema,
+  limitParamSchema,
+  offsetParamSchema,
+  validateBotonicEventData,
+} from './validation/events'
 
 const router = Router()
 
 router
   .route('/')
-  .get((_, res) => {
-    res.send('GET HTTP method on event resource')
-  })
-  .post((_, res) => {
-    res.send('POST HTTP method on event resource')
+  .get(
+    checkSchema({
+      limit: limitParamSchema,
+      offset: offsetParamSchema,
+    }),
+    async (req, res) => {
+      try {
+        const errors = validationResult(req)
+        if (!errors.isEmpty()) {
+          res.status(400).send({ errors: errors.array() })
+          return
+        }
+
+        const query = matchedData(req, { locations: ['query'] })
+        const limit = query.limit ?? undefined
+        const offset = query.offset ?? undefined
```

Comment:
Question here because I'm not sure: isn't this redundant? When `query.limit` is not defined, it will return `undefined`, right?

Reply:
Yes, you're right, the `??` operator here is redundant. Removed!